napari/components/dims.py

```diff
@@ -182,8 +206,7 @@ def set_range(self, axis: int, range: Sequence[Union[int, float]]):
         range : tuple
             Range specified as (min, max, step)
         """
-        if axis < 0:
-            axis += self.ndim
+        axis = axis % self.ndim
```

Comment:
do not use modulus here (and everywhere else) for negative indexing. instead do what you had before and raise an exception if it's still negative afterwards

```python
In [1]: l = [0, 1]

In [2]: l[-3]
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-2-55afb2c9d60b> in <module>
----> 1 l[-3]

IndexError: list index out of range
```

Reply:
The modulus will force everything to be positive though, why is this undesirable. In the example above you'd have
```python
l[-3 % 2]
```
which is equivalent to
```python
l[1]
```
I'm happy to change though as if someone passes `-17` and the list is of length 3 the probably deserve an error ....

Reply:
my point is that it is standard practice to only perform this wrap once instead of forcing it to be positive always

Reply:
got it - will fix